### PR TITLE
Support for links to the standard library.

### LINKED ISSUE #11: Rustdoc intra-links should be translated to links to doc.rs
An intra-link inside rustdoc will not work in the README.

Example:

If we have this in `lib.rs`:

```rust
//! +Archery defines a [`SharedPointer`](./shared_pointer/struct.SharedPointer.html)
```

the README should probably become:

```text
Archery defines a [`SharedPointer`](https://docs.rs/archery/latest/archery/shared_pointer/struct.SharedPointer.html)
```

### PR DESCRIPTION
**Pull request by orium**
Should be easier to review commit by commit, since they each one of them has a very concrete purpose.

Closes #11.

### CODE REVIEW

**Comment by orium** [comment]
Marked as draft because I just noticed a bug.  We are detecting links inside backticked stuff, which should be skipped.  It is not a trivial fix and there will likely be other edge cases, so I might take a look at a proper markdown parser to extract the links (I'm gonna take a loo at pulldown_cmark).

**Comment by orium** [comment]
`pulldown_cmark` solved the misidentified link inside code blocks and I could get rid of that ugly and complex link parser.  Ready for review :slightly_smiling_face:

File: src/intralinks.rs [hunk 1]
```diff
@@ -185,38 +186,73 @@ impl Debug for FQIdentifier {
...
+  // We choose to consider the first one only.
+  match asts.contains_key(&mod_symbol) {
+    true => return Ok(()),
+    false => asts.insert(mod_symbol.clone(), ast.clone()),
+  };
 
   for item in ast.iter() {
     if let Item::Mod(module) = item {
-      let module_symbol: FQIdentifier = mod_symbol.clone().join(&module.ident.to_string());
+      // If a module is gated by `#[cfg(test)]` we skip it.  This is happens sometimes in the
```
Comment (phaazon) [inline src/intralinks.rs:224]:
```suggestion
      // If a module is gated by `#[cfg(test)]` we skip it.  This happens sometimes in the
```

**Review by phaazon — CHANGES_REQUESTED** [review]

**Review by phaazon — APPROVED** [review]

**Comment by orium** [comment]
Thanks for merging this @phaazon.  Just a small thing about the [CHANGELOG.md](https://github.com/phaazon/cargo-sync-readme/blob/master/CHANGELOG.md#11), where is

> This allows to point correctly to the right crates.io link

there should be

> This allows to point correctly to the right docs.rs link

Also, it is worth mentioning that links to the standard library will also point to the documentation in doc.rust-lang.org

**Pull request merged**